validation/static/editSegment.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+document.addEventListener('DOMContentLoaded', () => {
+    for (let button of document.querySelectorAll('[data-segment-name]')) {
+        button.addEventListener("click", async (e) => {
+            showDiv()
+            const name = e.target.dataset.segmentName
+            const translation = document.getElementById(name + '-translation').innerHTML;
+
+            document.getElementById('id_cree').value = name;
+            document.getElementById('id_translation').value = translation;
+        })
+    }
+
+    for (let button of document.querySelectorAll('[data-cy="accept-button"], [data-cy="revert-button"]')) {
+        button.addEventListener("click", async(e) => {
+            showDiv()
+
+            const transcription = e.target.dataset.transcription
+            const translation = e.target.dataset.translation
+            const analysis = e.target.dataset.analysis
```

Comment:
If you want to use some [fancier syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)*

```suggestion
            const { transcription, translation, analysis } =  e.target.dataset;
```
<hr>
*I think this has now has excellent [browser support](https://caniuse.com/?search=destructuring)